python_modules/dagit/dagit/pipeline_run_storage.py

```diff
@@ -50,6 +52,29 @@ def __init__(self, run_id, pipeline_name, config, execution_plan):
         self.pipeline_name = pipeline_name
         self.config = config
         self.execution_plan = execution_plan
+        self._log_queue_lock = gevent.lock.Semaphore()
+        self._log_queue = []
+        self._flush_queued = False
+        self._queue_timeout = None
+
+    def _enqueue_flush_logs(self):
+        if not self._flush_queued:
```

Comment:
let's put _flushed_queued behind a lock. in particular I think we need to test for false and set to true atomically